recipes/samarium/v1.0.1/conanfile.py

```diff
@@ -0,0 +1,85 @@
+from os import path
```

Comment:
Why do you need an entire new folder for a patch version? It sounds odd.

Reply:
Maybe I've misunderstood, but `v1.0.0` used conans imports, different generators, and hence different cmake files.

Can patches also be used for the `conanfile.py` itself?

Reply:
Patches can be re-used for any version, since they are compatible. The recipe for Boost, re-uses a bunch of patches for many different versions.

If this new version is a patch, I see no reason for breaking behavior of changing things. Keep the same folder for both versions, please.

Reply:
Ok thanks, will do.